sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -629,22 +629,56 @@ def estimate_size(self, unused_value, nested=False):
 
 class TimerCoderImpl(StreamCoderImpl):
   """For internal use only; no backwards-compatibility guarantees."""
-  def __init__(self, payload_coder_impl):
+  def __init__(self, key_coder_impl, window_coder_impl, tag_coder_impl):
     self._timestamp_coder_impl = TimestampCoderImpl()
-    self._payload_coder_impl = payload_coder_impl
+    self._boolean_coder_impl = BooleanCoderImpl()
+    self._pane_info_coder_impl = PaneInfoCoderImpl()
+    self._key_coder_impl = key_coder_impl
+    self._windows_coder_impl = TupleSequenceCoderImpl(window_coder_impl)
+    self._tag_coder_impl = tag_coder_impl
 
   def encode_to_stream(self, value, out, nested):
     # type: (dict, create_OutputStream, bool) -> None
-    self._timestamp_coder_impl.encode_to_stream(value['timestamp'], out, True)
-    self._payload_coder_impl.encode_to_stream(value.get('payload'), out, True)
+    self._key_coder_impl.encode_to_stream(value.user_key, out, nested)
```

Comment:
```suggestion
    self._key_coder_impl.encode_to_stream(value.user_key, out, True)
```

We should ignore the nested field since it will change the encoding.